### ISSUE TITLE
Malformed repo string causes error

### ISSUE THREAD

**Issue by rferris** [furlongm/patchman#65]
One of the EPEL mirrors sent via patchman-client is not a full repo string:
```
'rpm' 'http://mirror.overthewire.com.au/pub/epel/7/x86_64' 
```
This causes the following:
```
Traceback (most recent call last):
  File "/usr/bin/patchman", line 4, in <module>
    __import__('pkg_resources').run_script('patchman==1.0.17', 'patchman')
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1438, in run_script
    exec(code, namespace, namespace)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/EGG-INFO/scripts/patchman", line 593, in <module>
    main()
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/EGG-INFO/scripts/patchman", line 588, in main
    showhelp = process_args(args)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/EGG-INFO/scripts/patchman", line 555, in process_args
    process_reports(args.host, args.force)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/EGG-INFO/scripts/patchman", line 430, in process_reports
    report.process(find_updates=False)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/patchman/reports/models.py", line 151, in process
    process_repos(report=self, host=host)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/patchman/reports/utils.py", line 43, in process_repos
    repo, priority = process_repo(repo_str, report.arch)
  File "/usr/lib/python2.7/site-packages/patchman-1.0.17-py2.7.egg/patchman/reports/utils.py", line 213, in process_repo
    if repo[2] == '':
IndexError: list index out of range
```

My quick fix for this was to modify the process_repos function in patchman/reports/utils.py as follows:
```
# Line 43
if len(repo_str) > 2:
                repo, priority = process_repo(repo_str, report.arch)
            else:
                repo = None
```

This just skips the repo and allows my report to succeed. I've seen mention in previous issues that this is the fault of the patchman-client, but perhaps a check for a properly formed repo string in the patchman server isn't such a bad idea?

**Comment by furlongm** [comment]
Looks like it's not correctly parsing the Repo-baseurl if Repo-metalink exists too. This is a regression in the client due to 1b151d581888647cbb6665383f92840b2e90a10b

It should ignore Repo-baseurl if Repo-metalink exists.

**Comment by furlongm** [comment]
The client should be fixed in 96525d06884c17ad45e889aa9ce8caeb099ee7bc

**Comment by rferris** [comment]
Awesome. Love your work.
